nixos/modules/config/swap.nix

```diff
@@ -157,11 +185,11 @@ let
 
     };
 
-    config = rec {
+    config = {
       device = mkIf options.label.isDefined
         "/dev/disk/by-label/${config.label}";
       deviceName = lib.replaceStrings ["\\"] [""] (escapeSystemdPath config.device);
-      realDevice = if config.randomEncryption.enable then "/dev/mapper/${deviceName}" else config.device;
+      realDevice = if config.randomEncryption.enable then "/dev/mapper/${config.deviceName}" else config.device;
```

Comment:
This change allows the user to `mkForce` the value of `deviceName` elsewhere in the configuration rather than binding the value set here.

The rationale for introducing this change is that if one specifies the device by using a path under `/dev/disk/by-id`, `/dev/disk/by-partuuid`, or `/dev/disk/by-partlabel` then the resulting `deviceName` becomes rather unwieldy.